File: source/backend/RandomReads/controllers/UserActivity.cs
using RandomReads.Models;
/*
User activity where - it stores the 
user as pkey
readid as id

{
user - user1
id - read1
topic
iscompleted - true
timespent - 100
isshared - false
isliked - true
isreported - true
}
------------------------------------------
TopicActivity
Topic - pKey
readid as id
{
topic
id - readid
likes
reports
timespent
shares

}
*/

public class UserActivity
{
    public string userid {get; set;} //pk
    public Topic topic {get; set;}
    public string readid {get; set;} //id
    public bool iscompleted {get; set;}
    public int timespent {get; set;}
    public bool islike {get; set;}
    public bool ishared {get; set;} 
    public bool isreported {get; set;} 
}

File: source/app/randomreads/lib/models/user_activity.dart
class UserActivity {
  final String userid;
  final String topic;
  final String readid;

  int timeSpent;        // in ms
  bool isCompleted;
  bool isLiked;
  bool isShared;
  bool isReported;

  UserActivity({
    required this.userid,
    required this.topic,
    required this.readid,
    this.timeSpent = 0,
    this.isCompleted = false,
    this.isLiked = false,
    this.isShared = false,
    this.isReported = false
  });

  Map<String, dynamic> toJson() => {
        'userid': userid,
        'topic': topic,
        'readid': readid,
        'timeSpent': timeSpent,
        'isCompleted': isCompleted,
        'islike': isLiked,
        'isshared': isShared,
        'isreported': isReported
      };

  factory UserActivity.fromJson(Map<String, dynamic> json) {
    return UserActivity(
      userid: json['userid'],
      topic: json['topic'],
      readid: json['readid'],
      timeSpent: json['timeSpent'] ?? 0,
      isCompleted: json['isCompleted'] ?? false,
      isLiked: json['isLiked'] ?? false,
      isShared: json['isShared'] ?? false,
      isReported: json['isReported'] ?? false,
    );
  }
}
